DM-50473: Write felis/sdm_schemas section for pipelines paper (#46)
lsst-pst/pstn-019: schemas.tex

\subsection{Catalog Schemas}
\label{sec:schemas}

Pipeline products must be transformed from the internal data model to the public data model defined in \citet{LSE-163}.
A set of YAML files in the \texttt{pipe\_tasks}\footnote{\url{https://github.com/lsst/pipe\_tasks}} repository are used for transforming the internal pipelines representation of the data to a standardized parquet output format.

> These parquet files are continually validated against an appropriate schema to ensure that the column names and types are correct as the pipelines codebase evolves, through repositories such as \texttt{ci\_imsim}\footnote{\url{https://github.com/lsst/ci\_imsim}} and \texttt{ci\_hsc}\footnote{\url{https://github.com/lsst/ci\_hsc}}.

Reviewer: Maybe we cite section 9.3 rather than listing explicit CI packages, and then we expand section 9.3 to distinguish nightly CI from CI of larger datsaets.

Reviewer: Did you forget to push something? You thumbsupped but the sentence didn't change.

Author: Changed to this:

```
These parquet files are continually validated against an appropriate schema to ensure that the column names and types are correct as the pipelines codebase evolves, as described in Section~\ref{sec:validation}.
```

Author: I added a reference to section 9.3.

> then we expand section 9.3 to distinguish nightly CI from CI of larger datsaets

Did you want me to add this on this PR?

Reviewer: If you could put a "Mention CI packages like ci\_hsc here, as referenced from schema section" in the validation section that would be sufficient.

For data previews and releases, the parquet files are then ingested into the Qserv database, where the data catalogs are stored.

> The public data model is defined by a set of files in the Felis\citep{2024arXiv241209721M} YAML format which describe the schema of a data catalog, including its tables, columns, constraints and metadata.

Reviewer: ```suggestion
The public data model is defined by a set of files in the Felis \citep{2024arXiv241209721M} YAML format which describe the schema of a data catalog, including its tables, columns, constraints and metadata.
```

> The YAML files are managed with the \texttt{sdm\_schemas} repository\footnote{\url{https://github.com/lsst/sdm\_schemas}} with all changes validated by GitHub workflows.

Reviewer: Somewhere we should note that "SDM" means "Science Data Model". Maybe in the second line above.

The schemas corresponding to Science Pipelines output are continually evolving with the pipelines codebase, so, for instance, column names and types may be updated to reflect changes to the internal data model.
Schemas for data previews and releases represent a snapshot of the public data model at the time of the release and would typically only be updated with bug fixes, minor changes, or updates and additions to the metadata.

> For public-facing data catalogs, the Felis representation is used to generate a TAP\_SCHEMA database describing the tables and columns available in the TAP\footnote{\url{https://www.ivoa.net/documents/TAP/}} service and serves as a source of documentation.

Reviewer: ```suggestion
For public-facing data catalogs, the Felis representation is used to generate a TAP\_SCHEMA database describing the tables and columns available in the TAP service \citep{2019ivoa.spec.0927D} and serves as a source of documentation.
```
Don't use footnotes for citations.

Document is officially at https://ui.adsabs.harvard.edu/abs/2019ivoa.spec.0927D/abstract

> User data access to the data catalogs is provided primarily through the Astronomical Data Query Language (ADQL) through the TAP service, with the tables and columns being validated against the TAP\_SCHEMA as part of the query execution process.

Reviewer: ```suggestion
User data access to the data catalogs is provided primarily through the Astronomical Data Query Language \citep[ADQL,][]{2023ivoa.spec.1215M} through the TAP service, with the tables and columns being validated against the TAP\_SCHEMA as part of the query execution process.
```

https://ui.adsabs.harvard.edu/abs/2023ivoa.spec.1215M/abstract
